board: MIMXRT1062-FMURT6: Add pinmux support for uart ports used for gps and telemtry.

sumitbatra-nxp opened the pull request:
1) This patch configures the correct pinmux settings for -

     lpuart2, lpuart3, lpuart5 ports used for gps and telemetry modules.

2) It also adds software pull up for the UART console port lpuart7 to 

3) It fixes an issue in ref clock configuration for ETH2 module

4) It also increases the RAM size

Fixes #59005 
Fixes #59010 
Fixes #59013

sumitbatra-nxp committed:
board: MIMXRT1062-FMURT6: Add pinmux support for uart ports

used for gps and telemtry.

This patch configures the correct pinmux settings for

lpuart2, lpuart3, lpuart5 ports used for gps and telemetry modules.

It also adds software pull up for the UART console port lpuart7.

Signed-off-by: Sumit Batra <sumit.batra@nxp.com>
Co-authored-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>
Co-authored-by: Iain Galloway <iaintravelgalloway@gmail.com>

sumitbatra-nxp committed:
board: MIMXRT1062-FMURT6: Removing the CONFIG_DISPLAY check in CMakeLists

This patch removes an unncessary message for presence or absence of DISPLAY

Signed-off-by: Sumit Batra <sumit.batra@nxp.com>

danieldegrasse commented on boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts:
```diff
@@ -33,6 +33,9 @@
 		zephyr,console = &lpuart7;
 		zephyr,shell-uart = &lpuart7;
 		zephyr,canbus = &flexcan1;
+		zephyr,gps1 = &lpuart2;
```
Are these well known Zephyr chosen node names? I don't see them being used anywhere in the tree. I would rather see these node names be devicetree aliases if these UART connections are being used in a downstream project

sumitbatra-nxp replied:
Changed from Chosen to Aliases as suggested

danieldegrasse commented on drivers/ethernet/eth_mcux.c:
```diff
@@ -1059,6 +1059,12 @@ static void eth_mcux_init(const struct device *dev)
+	/* Set ENET2 ref clock to be generated by External OSC,*/
+	/* direction as output and frequency to 50MHz */
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(enet2), okay)
+	IOMUXC_GPR->GPR1 |= (IOMUXC_GPR_GPR1_ENET2_CLK_SEL_MASK |
```
This looks like it should be handled at the `soc_rt10xx.c` level, not within the driver. Also, is there an `IOMUXC` API present for this bit, like `kIOMUXC_GPR_ENET1TxClkOutputDir` for ENET1?

sumitbatra-nxp replied:
Thanks @danieldegrasse .. I have modified as per your recommendation.. please check now..

sumitbatra-nxp committed:
soc: rt10xx: fix the sequence of Enet2 ref clk enablement

This patch sets ENET2 ref clock to be generated by External OSC

ENET2 ref clock direction as output

ENET2 ref clk frequency to 50MHz

Signed-off-by: Sumit Batra <sumit.batra@nxp.com>

sumitbatra-nxp commented:
@danieldegrasse .. can you please re-check.. 
@dleach02.. @mmahadevan108  can you please review

bperseghetti commented:
@danieldegrasse we added the issues for the bug report fixs and linked them.

danieldegrasse commented on boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts:
```diff
@@ -20,12 +20,39 @@
 		telem4-gps2 = &lpuart5;
 	};
 
+	pwm_shell: pwm_shell {
```
the commit title `MIMXRT1062-FMURT6: Fix PWM device names.` Should be updated to something like the following:
`soc: arm: nxp_imx: fix PWM node names`

bperseghetti committed:
soc: nxp_rt10xx: add unique PWM names.

Enable PWM to use unique device names.

Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

sumitbatra-nxp committed:
board: MIMXRT1062-FMURT6: Add pinmux support for FMU PPM Input

This commit adds the Pinmux and Pad settings for FMU PPM Input Pin

Signed-off-by: Sumit Batra <sumit.batra@nxp.com>

sumitbatra-nxp committed:
board: MIMXRT1062-FMURT6: Add pinmux support for I2C3 module Pins

This patch adds the pinmux and pad settings for I2C3 SDA and SCL pins

Signed-off-by: Sumit Batra <sumit.batra@nxp.com>

danieldegrasse commented on soc/arm/nxp_imx/rt/soc_rt10xx.c:
```diff
@@ -66,7 +66,12 @@ const clock_enet_pll_config_t ethPllConfig = {
 #else
 	.enableClkOutput25M = false,
 #endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(enet), okay)
```
Commit message here needs to be to something like- `soc: rt10xx: fix the sequence of Enet2 ref clk enablement`

bperseghetti replied:
Commit title/message now updated.

bperseghetti committed:
board: MIMXRT1062-FMURT6: correct PWM prescaler

Set the PWM clock prescaler values to 64 for controlling standard servo
50Hz carrier frequency.

Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

bperseghetti committed:
board: MIMXRT1062-FMURT6: Set CANFD bus-speed-data

Set the flexcan3 bus-speed-data to allow for correct CANFD data-speed.

Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

bperseghetti commented on boards/arm/mimxrt1062_fmurt6/mimxrt1062_fmurt6.dts:
```diff
@@ -143,6 +142,7 @@
 	pinctrl-0 = <&pinmux_flexcan3>;
 	pinctrl-names = "default";
 	bus-speed = <125000>;
+	bus-speed-data = <1000000>;
```
Done. Split it out into:
- board: MIMXRT1062-FMURT6: Set CANFD bus-speed-data
- board: MIMXRT1062-FMURT6: correct PWM prescaler

and renamed it to be more proper with:
- soc: nxp_rt10xx: add unique PWM names.

sumitbatra-nxp commented:
@mmahadevan108 .. Can you kindly review, as we need a second approver as well.
